homeassistant/components/life360/device_tracker.py

```diff
@@ -1,432 +1,267 @@
-    CONF_ERROR_THRESHOLD,
     CONF_MAX_GPS_ACCURACY,
-    CONF_MAX_UPDATE_WAIT,
-    CONF_MEMBERS,
-    CONF_SHOW_AS_STATE,
-    CONF_WARNING_THRESHOLD,
     DOMAIN,
+    LOGGER,
     SHOW_DRIVING,
-    SHOW_MOVING,
 )
 
-_LOGGER = logging.getLogger(__name__)
+_EXTRA_ATTRIBUTES = (
+    ATTR_ADDRESS,
+    ATTR_AT_LOC_SINCE,
+    ATTR_BATTERY_CHARGING,
+    ATTR_LAST_SEEN,
+    ATTR_PLACE,
+    ATTR_SPEED,
+    ATTR_WIFI_ON,
+)
 
-SPEED_FACTOR_MPH = 2.25
-EVENT_DELAY = timedelta(seconds=30)
 
-ATTR_ADDRESS = "address"
-ATTR_AT_LOC_SINCE = "at_loc_since"
-ATTR_DRIVING = "driving"
-ATTR_LAST_SEEN = "last_seen"
-ATTR_MOVING = "moving"
-ATTR_PLACE = "place"
-ATTR_RAW_SPEED = "raw_speed"
-ATTR_SPEED = "speed"
-ATTR_WAIT = "wait"
-ATTR_WIFI_ON = "wifi_on"
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the device tracker platform."""
+    account = hass.data[DOMAIN]["accounts"][entry.unique_id]
+    coordinator = account["coordinator"]
+    tracked_members = hass.data[DOMAIN]["tracked_members"]
+    logged_circles = hass.data[DOMAIN]["logged_circles"]
+    logged_places = hass.data[DOMAIN]["logged_places"]
+
+    @callback
+    def process_data(new_members_only: bool = True) -> None:
+        """Process new Life360 data."""
+        for circle_id, circle in coordinator.data["circles"].items():
+            if circle_id not in logged_circles:
+                logged_circles.append(circle_id)
+                LOGGER.info("Circle: %s", circle["name"])
+
+            new_places = []
+            for place_id, place in circle["places"].items():
+                if place_id not in logged_places:
+                    logged_places.append(place_id)
+                    new_places.append(place)
+            if new_places:
+                msg = f"Places from {circle['name']}:"
+                for place in new_places:
+                    msg += f"\n- name: {place['name']}"
+                    msg += f"\n  latitude: {place['latitude']}"
+                    msg += f"\n  longitude: {place['longitude']}"
+                    msg += f"\n  radius: {place['radius']}"
+                LOGGER.debug(msg)
+
+        new_entities = []
+        for member_id, member in coordinator.data["members"].items():
+            tracked_by_account = tracked_members.get(member_id)
+            if new_member := not tracked_by_account:
+                tracked_members[member_id] = entry.unique_id
+                LOGGER.info("Member: %s", member[ATTR_NAME])
+            if (
+                new_member
+                or tracked_by_account == entry.unique_id
+                and not new_members_only
+            ):
+                new_entities.append(Life360DeviceTracker(coordinator, member_id))
+        if new_entities:
+            async_add_entities(new_entities)
+
+    process_data(new_members_only=False)
+    account["unsub"] = coordinator.async_add_listener(process_data)
```

Comment:
Please use `config_entry.async_on_unload` to handle unsubscribing listeners on config entry unload.

Reply:
Ok.